src/toxcore/stats.rs

```diff
@@ -0,0 +1,112 @@
+/*!
+Statistics of incoming/outgoing packets
+This is used by both Udp codec and Tcp codec.
+*/
+
+#[cfg(target_pointer_width = "64")]
+use std::sync::atomic::*;
+#[cfg(target_pointer_width = "32")]
+use std::sync::Mutex;
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Default)]
+/// Struct for counting packets on x64 CPU.
+#[cfg(target_pointer_width = "64")]
+pub struct Stats {
+    /// Incoming packets count for Udp/Tcp
+    incoming: Arc<AtomicUsize>,
```

Comment:
I'd use one arc per Stats struct. Just pass Arc<Stats> to Codec.

Reply:
I remember that you said "separated encoding, decoding would be better". So two lock is better

Reply:
Why? I like the current implementation

Reply:
But `Arc` is not a lock - it's necessary for memory management only.

Reply:
> Why? I like the current implementation

If we need to add new counter we will have to wrap every counter to Arc. But it's absolutely unnecessary - we can have only one Arc for all counters.

Reply:
In the future we can change the inner implementation to:
```
struct Stats {
   counters: Arc<Counters>
}
struct Counters {
   ...
}
```

No need to move Arc into codecs

Reply:
Then why not to do it now? We already have two counters.

Reply:
OK.

Reply:
Moving Arc make it impossible to clone.

Reply:
Codec should be clonable

Reply:
They will be clonable. Don't worry

Reply:
done